### DISCUSSION TITLE
About the source code

### DISCUSSION THREAD

**Discussion by JAJA-WHB** [audacity/audacity#2856] [Development]
Which part of the source code is high pass filtering?
And which part is the denoise?
Colud you please tell me?

**Comment by Pokechu22** [comment]
I believe high-pass filtering is:

https://github.com/audacity/audacity/blob/74b333b2670c7a41efe24b7e76ca73b95dd388a4/plug-ins/highpass.ny

which calls into this:

https://github.com/audacity/audacity/blob/74b333b2670c7a41efe24b7e76ca73b95dd388a4/nyquist/dspprims.lsp#L21-L33

but I'm not entirely sure where things go from there.

**Comment by Pokechu22** [comment]
And there seem to be two denoise implementations: https://github.com/audacity/audacity/blob/74b333b2670c7a41efe24b7e76ca73b95dd388a4/src/effects/NoiseRemoval.cpp and https://github.com/audacity/audacity/blob/74b333b2670c7a41efe24b7e76ca73b95dd388a4/src/effects/NoiseReduction.cpp (with `NoiseReduction.cpp` being a bit newer, but I don't really know what the difference is between them)

**Reply by JAJA-WHB** [reply]
Can the high pass filtering code be implemented with data processing in Python?
I can't find a suitable way to achieve this.

**Reply by JAJA-WHB** [reply]
I've only recently started learning about audio noise reduction

**Reply by Pokechu22** [reply]
It definitely is possible to implement it in python, but I don't think the code that I linked would be useful for that; you will need to find lower-level code (it probably is somewhere in the Nyquist library Audacity uses).

**Reply by JAJA-WHB** [reply]
What compiler do you use to run LISP？

**Reply by JAJA-WHB** [reply]
clisp or sbcl?

**Reply by Paul-Licameli** [reply]
Audacity uses an old (from the 80s!) and simple Lisp interpreter in lib-src/Nyquist/xlisp which supplies bindings to signal processing functions implemented efficiently in C.

NoiseReduction.cpp has the current implementation of the effect.  NoiseRemoval is an older version that is not part of the build by default, but never got purged from the source tree.
